ArcGISRuntimeSDKQt_QMLSamples/Maps/DownloadPreplannedMap/DownloadPreplannedMap.qml

```diff
@@ -0,0 +1,284 @@
+                Layout.alignment: Qt.AlignHCenter
+                text: qsTr("Show Online Map")
+                enabled: !busy.visible & !viewingOnlineMaps
+                onClicked: {
+
+                    mapView.map = onlineMap;
+                    graphicsOverlay.visible = true;
+                    viewingOnlineMaps = true;
+                    mapView.setViewpointGeometry(offlineMapTask.preplannedMapAreaList.get(preplannedCombo.currentIndex).areaOfInterest);
+                    checkIfFileExists(preplannedCombo.currentIndex);
+                }
+            }
+
+            Text {
+                text: qsTr("Available Preplanned Areas:")
+                color: "white"
+                Layout.alignment: Qt.AlignHCenter
+                Layout.margins: 2
+            }
+
+            ComboBox {
+                id: preplannedCombo
+                Layout.fillWidth: true
+                Layout.margins: 2
+                enabled: !busy.visible
+                model: null
+                textRole: "itemTitle"
+
+                onActivated: {
+                    if (offlineMapTask.preplannedMapAreaList.count <= 0)
+                        return;
+
+                    if (offlineMapTask.preplannedMapAreaList.get(currentIndex).loadStatus !== Enums.LoadStatusLoaded)
+                        return;
+
+                    checkIfFileExists(currentIndex);
+
+                    if (viewingOnlineMaps)
+                        mapView.setViewpointGeometry(offlineMapTask.preplannedMapAreaList.get(currentIndex).areaOfInterest);
+                }
+            }
+
+            Button {
+                id: downloadOrView
+                Layout.fillWidth: true
+                Layout.margins: 2
+                enabled: !busy.visible
+                text: mapExists ? qsTr("View Preplanned area") : qsTr("Download Preplanned Area")
+                onClicked: {
+                    preplannedArea = offlineMapTask.preplannedMapAreaList.get(preplannedCombo.currentIndex);
+                    offlineMapTask.createDefaultDownloadPreplannedOfflineMapParameters(preplannedArea);
+                    graphicsOverlay.visible = false;
+                    viewingOnlineMaps = false;
+                    busy.visible = true;
+                }
+
+                onWidthChanged: {
+
+                }
+            }
+
+            Text {
+                text: qsTr("Download(s) deleted on exit")
+                color: "white"
+                Layout.alignment: Qt.AlignHCenter
+                Layout.margins: 2
+            }
+        }
+    }
+
+    FileFolder {
+        id: fileFolder
+    }
+
+    BusyIndicator {
+        id: busy
+        anchors.centerIn: parent
+        visible: false;
+    }
+
+    function checkIfFileExists(index) {
+        path = outputMapPackage + "/" + offlineMapTask.preplannedMapAreaList.get(index).portalItem.title + ".mmpk";
+        fileFolder.url = path;
+        mapExists = fileFolder.exists;
```

Comment:
the function name seems like it would return a bool, but it seems like it returns void and sets some variable bool. Maybe this can be cleaned up

Reply:
yeah I bounced between two ways of doing this and ended up having a little of both I see.